### PR TITLE
HDDS-10403. CopyObject should set ETag based on the key content

### PR DESCRIPTION
**Pull request by ivandika3**
## What changes were proposed in this pull request?

Currently copyObject does not seem to support eTag based on key's content. It uses a random UUID in CopyObjectResponse.

We need to change this to be the eTag based on the key content instead.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-10403

## How was this patch tested?

Unit tests, acceptance test, and manual tests.

Compatibilities
- If the source key does not contain eTag field, the copied destination key will have an eTag field

Clean CI run: https://github.com/ivandika3/ozone/actions/runs/8001415597

### CODE REVIEW

**Comment by ivandika3** [comment]
@kerneltime @myskov Could you help to review this when convenient? Thank you.

File: hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/ObjectEndpointStreaming.java [hunk 1]
```diff
@@ -122,15 +122,19 @@ public static long copyKeyWithStream(
...
       throws IOException {
     long writeLen = 0;
+    String eTag;
     try (OzoneDataStreamOutput streamOutput = bucket.createStreamKey(keyPath,
         length, replicationConfig, keyMetadata)) {
       long metadataLatencyNs =
           METRICS.updateCopyKeyMetadataStats(startNanos);
-      perf.appendMetaLatencyNanos(metadataLatencyNs);
       writeLen = writeToStreamOutput(streamOutput, body, bufferSize, length);
+      eTag = DatatypeConverter.printHexBinary(body.getMessageDigest().digest())
```
[the remark this answers is not in the document]
Comment (ivandika3) [reply hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/ObjectEndpointStreaming.java]: Thanks for the review. Updated. PTAL.

**Comment by adoroszlai** [comment]
Thanks @ivandika3 for the patch, @myskov, @vtutrinov for the review.

**Comment by ivandika3** [comment]
Thank you for the reviews @myskov @vtutrinov and @adoroszlai for the merge.

**Pull request merged**